src/components/Tree/Node.tsx

```diff
@@ -56,81 +91,98 @@ export const TreeNode = ({
+                        className="tw-flex tw-items-center tw-flex-grow tw-justify-between tw-cursor-pointer"
+                        aria-selected={selected}
+                        onClick={onNodeClick}
+                    >
+                        <div className="tw-flex tw-space-x-2 tw-items-center">
+                            <span data-test-id="toggle" onClick={toggleNodesVisibility}>
+                                {nodes &&
+                                    (showNodes ? (
+                                        <IconCaretDown size={IconSize.Size16} />
+                                    ) : (
+                                        <IconCaretRight size={IconSize.Size16} />
+                                    ))}
+                            </span>
+                            {icon && <span>{icon}</span>}
+                            <span data-test-id="node-link-name">{name}</span>
+                        </div>
+                        <div className="tw-px-1.5">
+                            <span
+                                className={merge([
+                                    "tw-text-black-100 tw-text-opacity-40 tw-font-normal",
+                                    selected && "tw-text-black-50",
+                                ])}
+                            >
+                                {label}
+                            </span>
+                        </div>
+                    </a>
+                    <AnimatePresence>
+                        {actions && (isHovered || selected) && (
+                            <motion.div
+                                className="tw-flex tw-space-x-1.5 tw-items-center"
+                                initial={{ width: 0 }}
+                                animate={{ width: "auto" }}
+                                exit={{ width: 0 }}
+                                transition={{ ease: [0.04, 0.62, 0.23, 0.98] }}
+                            >
+                                {actions.map((action) => action)}
+                            </motion.div>
+                        )}
+                    </AnimatePresence>
+                </div>
+            </DropZone>
 
             {nodes && showNodes && (
                 <ul
                     className="tw-p-0 tw-m-0 tw-font-sans tw-font-normal tw-list-none tw-text-left"
                     data-test-id="sub-tree"
                 >
-                    {nodes.map((node) => {
-                        return (
-                            <TreeNode
-                                key={node.id}
-                                node={node}
-                                activeNodeId={activeNodeId}
-                                onClick={onClick}
-                                parentIds={[...parentIds, id]}
-                            />
-                        );
-                    })}
+                    {renderNodesArray(nodes, activeNodeId, onClick, onDrop, [...parentIds, id])}
```

Comment:
renderNodesArray maybe better name should be renderNodes or renderNodeArray. As this way its redundant to call it nodes with array.